cpp/src/gandiva/decimal_xlarge.cc

```diff
@@ -38,7 +38,7 @@
 
 namespace gandiva {
 
-void ExportedDecimalFunctions::AddMappings(Engine* engine) const {
+arrow::Status ExportedDecimalFunctions::AddMappings(Engine* engine) const {
```

Comment:
For external C interface functions, they may cause some error when adding mapping for them, so I change the `AddGlobalMappings` and `ExportedFuncsBase::AddMappings` functions to return arrow::Status to represent the result